File: pomade-signer/src/pow.rs
/// Count the number of leading zero bits in a 32-byte hash.
pub fn get_pow(id: &[u8; 32]) -> u32 {
    let mut count = 0;

    for &byte in id {
        if byte == 0 {
            count += 8;
        } else {
            count += byte.leading_zeros();
            break;
        }
    }

    count
}

File: pomade-signer-go/pow.go
package main

func getPow(id [32]byte) uint32 {
	var count uint32
	for _, b := range id {
		if b == 0 {
			count += 8
			continue
		}
		for i := 7; i >= 0; i-- {
			if (b>>i)&1 == 0 {
				count++
				continue
			}
			return count
		}
	}
	return count
}
